libp2p/NodeTable.cpp

```diff
@@ -360,15 +360,14 @@ void NodeTable::noteActiveNode(shared_ptr<NodeEntry> _nodeEntry, bi::udp::endpoi
     }
     if (!isAllowedEndpoint(NodeIPEndpoint(_endpoint.address(), _endpoint.port(), _endpoint.port())))
     {
-        LOG(m_logger) << "Skipping making node with unallowed endpoint active. Node "
-                      << _nodeEntry->id << "@" << _endpoint;
+        LOG(m_logger) << "Skipping making node with unallowed endpoint active. Node " << dynamic_cast<Node const&>(*_nodeEntry);
```

Comment:
It should be `static_cast<>`, but there is a better way of fixing that. We have to add another overload of `operator<<` for `Node&` (without `const`). It is a known problem as the same solution was applied for other types.

It's defined here: https://github.com/ethereum/aleth/blob/master/libp2p/Common.h#L247.

Reply:
Ah, understood about the `static_cast` (static cast is preferred since we're casting to a non-virtual base class which means that it's a compile time operation). However, I don't understand why a better approach would be to overload the ostream operator for non-const `Node`& - why doesn't the existing overloaded `<< `work (for `const Node&`)?